sources/packages/backend/apps/api/src/route-controllers/designation-agreement/designation-agreement.aest.controller.ts

```diff
@@ -94,36 +98,40 @@ export class DesignationAgreementAESTController extends BaseController {
     @Param("designationStatus", new ParseEnumPipe(DesignationAgreementStatus))
     designationStatus: DesignationAgreementStatus,
     @Query(PaginationParams.SearchCriteria) searchCriteria: string,
```

Comment:
This query string is not validated and must be restricted in some way, we no longer use the `PaginationParams.SearchCriteria`. A DTO can be created to restrict the input, as below.
```ts
export class DesignationAgrementSearchAPIInDTO {
  /**
   * Criteria to be used to filter the records.
   */
  @IsOptional()
  @MaxLength(PAGINATION_SEARCH_MAX_LENGTH)
  searchCriteria?: string;
}
```

Reply:
OK. What should be the value for `PAGINATION_SEARCH_MAX_LENGTH`?

Reply:
The `PAGINATION_SEARCH_MAX_LENGTH` was already present but was defined based on first name/last name size. When I added the comment the idea was to reuse the existing const but I do not see a problem to have one specific for the designation agreement.